content/efcore/getting-started/manipulating-entities.md

```diff
@@ -0,0 +1,138 @@
+# Manipulating Entities 
+ 
+Four operations that we often perform on a record in a database are create, read, update, and delete (commonly referred to as "CRUD" operations). We walked through how to read records from a database in the [Querying Data](querying-data.md) tutorial. Now let's learn how to perform the other CRUD operations which allow us to manipulate records in the database. 
+ 
+## Adding Entities to a Database 
+ 
+Let's first look at how to create a new record in the database. In the following example, we create a new author and add it to the database: 
+  
+```{.snippet} 
+using (var context = new LibraryContext()) 
+{ 
+var author = new Author 
+{  
+FirstName = "Mary",  
+LastName = "Shelley", 
+Books = new List<Book>() 
+}; 
+context.Authors.Add(author); 
+context.SaveChanges(); 
+} 
+``` 
+:::repl{data-name=adding-entities} 
+::: 
+ 
+As you can see, it is easy to add new entities to the database with EF Core. First, we add the entity to the database using the `Add` method of the `DbSet`, much like we would when adding items to a `List`. Next, we call the context's `SaveChanges` method. This step is essential because rather than write changes to the database right away, EF Core keeps track of all the pending changes and waits to write them until `SaveChanges` is called. 
+ 
+Notice that we do not specify the `AuthorId` property before adding the author to the database. This is because `AuthorId` is a generated property, meaning EF Core will automatically generate a value for it. 
+ 
+## Adding Related Data 
+ 
+In the previous example, we added an author without any books to the database. Now we want to add an author and their books to the database. To add related entities like this to the database using EF Core we simply call the `Add` method on the parent entity. It will then automatically add the parent’s related entities to the database. In the following example, we add an author and their books to the database by simply adding the author: 
+ 
+``` 
+using (var context = new LibraryContext()) 
+{ 
+var author = new Author 
+{  
+FirstName = "Mary",  
+LastName = "Shelley", 
+Books = new List<Book> { 
+new Book 
+{ 
+Title = "Frankenstein: or, The Modern Prometheus", 
+Genre = "Science Fiction", 
+PublicationYear = 1818 
+ 
+} 
+} 
+}; 
+context.Authors.Add(author); 
+context.SaveChanges(); 
+} 
+``` 
+:::repl{data-name=adding-related-entities} 
+::: 
+ 
+If we want to add a new entity that is related to an existing entity, we first load the existing entity, then add the related entity to it. For example, to add a new book to an existing author we would load the author and then add the book to the author's list of books. Calling `SaveChanges` would then add the new book to the database. 
+ 
+``` 
+using (var context = new LibraryContext()) 
+{ 
+var author = context.Author 
+.Single(a => a.LastName == "Crane") 
+ 
+var book = new Book 
+{ 
+Title = "The Red Badge of Courage", 
```

Comment:
Gonna change this to _And Then There Were None_ by Agatha Christie, just so we don't have to deal with creating a new list for Crane and adding the book to it.